meinberlin/apps/kiezradar/serializers.py

```diff
@@ -0,0 +1,104 @@
+from rest_framework import serializers
+
+from adhocracy4.administrative_districts.models import AdministrativeDistrict
+from adhocracy4.projects.models import Topic
+from meinberlin.apps.kiezradar.models import KiezradarQuery
+from meinberlin.apps.kiezradar.models import ProjectType
+from meinberlin.apps.kiezradar.models import SearchProfile
+from meinberlin.apps.organisations.models import Organisation
+
+
+class SearchProfileSerializer(serializers.ModelSerializer):
+    """Serializer for the SearchProfile model."""
+
+    query = serializers.PrimaryKeyRelatedField(
+        queryset=KiezradarQuery.objects.all(), allow_null=True, required=False
+    )
+    organisations = serializers.PrimaryKeyRelatedField(
+        queryset=Organisation.objects.all(), many=True, allow_null=True, required=False
+    )
+    districts = serializers.PrimaryKeyRelatedField(
+        queryset=AdministrativeDistrict.objects.all(),
+        many=True,
+        allow_null=True,
+        required=False,
+    )
+    project_types = serializers.PrimaryKeyRelatedField(
+        queryset=ProjectType.objects.all(), many=True, allow_null=True, required=False
+    )
+    topics = serializers.PrimaryKeyRelatedField(
+        queryset=Topic.objects.all(), many=True, allow_null=True, required=False
+    )
+    status_display = serializers.CharField(source="get_status_display", read_only=True)
+
+    class Meta:
+        model = SearchProfile
+        fields = [
+            "id",
+            "user",
+            "name",
+            "description",
+            "disabled",
+            "status",
+            "status_display",
+            "query",
+            "organisations",
+            "districts",
+            "project_types",
+            "topics",
+        ]
+
+        read_only_fields = ["user"]
+
+    def create(self, validated_data):
+        # Pop many-to-many fields from validated_data
+        query = validated_data.pop("query", None)
+        organisations_data = validated_data.pop("organisations", None)
+        districts_data = validated_data.pop("districts", None)
+        project_types_data = validated_data.pop("project_types", None)
+        topics_data = validated_data.pop("topics", None)
+
+        search_profile = SearchProfile.objects.create(**validated_data)
+        if query:
+            query, _ = KiezradarQuery.objects.get_or_create(pk=query.id)
```

Comment:
I'm not sure I follow here: why use `get_or_create` here ? It only passes the pk, so the `text` field of the query will be empty, right?

Reply:
forgot to add the text.